DigitalLearningSolutions.Data/Services/TutorialContentService.cs

```diff
@@ -52,17 +46,22 @@ FROM Tutorials
                          INNER JOIN Applications
                          ON Customisations.ApplicationId = Applications.ApplicationId
 
-                         INNER JOIN Progress
-                         ON aspProgress.ProgressID = Progress.ProgressID
-                            AND CustomisationTutorials.CustomisationID = Progress.CustomisationID
-                   WHERE Progress.CandidateID = @candidateId
-                     AND CustomisationTutorials.CustomisationID = @customisationId
+                         LEFT JOIN Progress
+                         ON CustomisationTutorials.CustomisationID = Progress.CustomisationID
+                            AND Progress.CandidateID = @candidateId
+
+                         LEFT JOIN aspProgress
+                         ON aspProgress.TutorialID = Tutorials.TutorialID
+                            AND aspProgress.ProgressID = Progress.ProgressID
+
+                         LEFT JOIN TutStatus
+                         ON aspProgress.TutStat = TutStatus.TutStatusID
+                   WHERE CustomisationTutorials.CustomisationID = @customisationId
                      AND Tutorials.SectionId = @sectionId
                      AND Tutorials.TutorialID = @tutorialId
                      AND Customisations.Active = 1
                      AND CustomisationTutorials.Status = 1
-                     AND Progress.RemovedDate IS NULL
-                     AND Progress.SystemRefreshed = 0;",
+                     AND (Progress.CandidateID IS NULL OR (Progress.RemovedDate IS NULL AND Progress.SystemRefreshed = 0));",

```

Comment:
You did these checks in the join with progress I believe, in the query for the initial menu. Any reason for doing it differently here? 🙂

Reply:
I had forgotten I had did that, and doing this in the join would be nicer here I think, so I've changed that in b8e8702 🙂